xml/System.IO/FileSystemWatcher.xml

```diff
@@ -837,6 +862,7 @@
         <exception cref="T:System.ObjectDisposedException">The <see cref="T:System.IO.FileSystemWatcher" /> object has been disposed.</exception>
         <exception cref="T:System.PlatformNotSupportedException">The current operating system is not Microsoft Windows NT or later.</exception>
         <exception cref="T:System.IO.FileNotFoundException">The directory specified in <see cref="P:System.IO.FileSystemWatcher.Path" /> could not be found.</exception>
+        <exception cref="T:System.IO.IOException">Unix-based operating systems only: The configured user limit on the number of <c>inotify</c> instances has been reached, or the per-process limit on the number of open file descriptors has been reached.</exception>
```

Comment:
This exception is Linux specific. `Unix` -> `Linux`.

Reply:
We have seen reports of this happening on BSD systems.

Reply:
Does BSD support `inotify`? I thought it was Linux specific.

Reply:
I have no first-hand experience but that is what other people complain about.  I suppose Apple may have ported it.

Reply:
To be sure, do you have a reference to such a report?

Reply:
@Jozkee shared it with me and then deleted their comment.

Reply:
@yecril71pl I linked back to it in my next reply, but here's again https://github.com/dotnet/aspnetcore/issues/8449#issuecomment-514089725.

Reply:
> I linked back to it in my next reply, but here's again https://github.com/dotnet/aspnetcore/issues/8449#issuecomment-514089725.

This is about Linux running in a VM on Mac for docker. It is not about BSD.

Reply:
Looking at the [csproj](https://github.com/dotnet/runtime/blob/135a9608c411f89f947837701ab1fd7ff1d3a5d6/src/libraries/System.IO.FileSystem.Watcher/src/System.IO.FileSystem.Watcher.csproj#L72-L75) there are two platforms that can throw this exception: Linux and FreeBSD.

The FreeBSD part was added in https://github.com/dotnet/corefx/pull/42488. It uses libinotify.

libinotify has a limit of [2147483646](https://github.com/libinotify-kqueue/libinotify-kqueue/blob/ea7835fcafc3cee2a0d6c0e3c8034962c48f6afe/sys/inotify.h#L35) instances by default.

Reply:
> This is about Linux running in a VM on Mac for docker. It is not about BSD.

The next comment is about BSD.

Reply:
> The next comment is about BSD.

I don't see it, and when I search BSD on that page my browser doesn't see it either.

Based on the code, besides Linux only FreeBSD can throw the exception, and given the 2G default limit in libinotify, it isn't really limited.

Reply:
> > The next comment is about BSD.

> I don't see it, and when I search BSD on that page my browser doesn't see it either.

> `sysctl` doesn't seem to work the same way on mac

Reply:
This exception doesn't get thrown on mac, so we don't need to provide instructions for it.

Reply:
Forgive my being dumb&mdash;but for what other reason would the commenter complain about `sysctl` in a context where it is recommended to be used to deal with this exception?

Reply:
The exception occurs on a mac host that runs a Linux VM.
The commands need to be executed in the Linux VM, and not on the host (where they give an error).